tooling/src/main/java/org/opencds/cqf/tooling/casereporting/tes/TESPackageGenerator.java

```diff
@@ -0,0 +1,615 @@
+                    }
+
+                    if (relevantConditionGrouper.getCompose().getInclude().stream().noneMatch(i -> i.getValueSet().contains(new CanonicalType(reportingSpecificationGrouper.getUrl())))) {
+                        StringBuilder reference = new StringBuilder(reportingSpecificationGrouper.getUrl());
+                        if (reportingSpecificationGrouper.hasVersion() && !reportingSpecificationGrouper.getVersion().isEmpty() && !reportingSpecificationGrouper.getVersion().isBlank()) {
+                            reference.append("|").append(reportingSpecificationGrouper.getVersion());
+                        }
+                        relevantConditionGrouper.getCompose()
+                            .addInclude(new ValueSet.ConceptSetComponent().addValueSet(reference.toString()));
+                    }
+                }
+            }
+        }
+    }
+
+    private Bundle buildAndWriteBundle(List<IBaseResource> resourcesToBundle) {//Library manifest, List<ValueSet> conditionGroupers, List<ValueSet> inputBundleResources) {
+        BundleBuilder builder = new BundleBuilder(this.fhirContext);
+        builder.setBundleField("id", "tes-content-bundle");
+        for (IBaseResource resourceToBundle : resourcesToBundle) {
+            builder.addTransactionUpdateEntry(resourceToBundle);
+        }
+
+        Bundle bundle = (Bundle)builder.getBundle();
+        for (IOUtils.Encoding encoding : getOutputFileEncodings()) {
+            IOUtils.writeResource(bundle, this.getOutputPath(), encoding, FhirContext.forR4Cached());
+        }
+
+        return bundle;
+    }
+
+    private List<ConditionGroupingEntry> loadConditionGroupingDefinitions(String pathToConditionGrouperWorkbook) {
+        List<ConditionGroupingEntry> conditionGroupingEntries = new ArrayList<>();
+        try {
+            Workbook workbook = SpreadsheetHelper.getWorkbook(pathToConditionGrouperWorkbook);
+            conditionGroupingEntries = processConditionGroupingsSheet(workbook);
+            return conditionGroupingEntries;
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        return conditionGroupingEntries;
+    }
+
+    private List<ConditionGroupingEntry> processConditionGroupingsSheet(Workbook workbook) {
+        List<ConditionGroupingEntry> conditionGroupingEntries = new ArrayList<>();
+
+        Iterator<Row> rowIterator;
+
+        Sheet groupingCodesSheet = workbook.getSheetAt(CONDITIONGROUPINGSSHEETINDEX);
+        rowIterator = groupingCodesSheet.iterator();
+        // Skip header row
+        rowIterator.next();
+
+        while (rowIterator.hasNext()) {
+            Row row = rowIterator.next();
+
+            String conditionGroupingName = SpreadsheetHelper.getCellAsString(row, CONDITIONGROUPINGTITLEINDEX)
+//                .replace("\u00a0"," ")
+//                .replace("\u202F"," ")
+//                .replace("\u2019"," ")
+                .trim();
```

Comment:
Note that getCellAsString can return a null value. May want to use Objects.requireNonNull(SpreadsheetHelper.getCellAsString(row, CONDITIONGROUPINGTITLEINDEX)).trim() to avoid possible null pointer exceptions here and subsequent uses of the getCellAsString method